### PR TITLE
fix(shares): reply to message with attachments

### LINKED ISSUE #11239: Mentions in captions don't trigger notifications
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Write a message with a mention
2. Drop a file so the message turns into a caption
3. Hit send

### Expected behaviour
- 🔔 Mentioned users have a notification

### Actual behaviour
- 🔕 Mentioned users have no notification
- Also applies to replies after https://github.com/nextcloud/spreed/pull/11224

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* allows to reply to other messages with file share
* Fix #11239 

### 🖼️ Screenshots / Screencasts

|  🏡 Flow |
|------------|
Screenshot from 2023-12-14 09-41-10
image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team

---

### 🚧 Tasks

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not possible
- [ ] 📘 API documentation in `docs/` has been updated or is not required
- [ ] 🔖 Capability is added or not needed

### CODE REVIEW

**Comment by DorraJaouad** [comment]
don't forget the OoO
image

**Comment by DorraJaouad** [comment]
> Make this a feature request?

I think so, replying with a picture with caption. For now, when you intend to reply and insert caption with pic, quote is removed and it is send as a normal caption message.

The quote should be removed from NewMessage after sending 
image

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Actually we should just make this work, not hide it

File: lib/Chat/SystemMessage/Listener.php [hunk 1]
```diff
@@ -455,12 +455,14 @@ protected function sendSystemMessage(Room $room, string $message, array $paramet
 			$referenceId = (string) $referenceId;
 		}
 
+		$parent = $parameters['metaData']['replyTo'] ?? null;
```
Comment (nickvergessen) [inline lib/Chat/SystemMessage/Listener.php:458]: This needs validation, but I will make a commit

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
Backporting as we need to also support replying in #11239

**Review by Antreesy — COMMENTED** [review]
Tested Notifications API patch :green_circle: 

image
image

File: src/components/NewMessage/NewMessage.vue [hunk 2]
```diff
@@ -515,8 +515,9 @@ export default {
 			}
 
 			if (this.upload) {
-				// Clear input content from store
+				// Clear input content from store and remove Quote component
 				this.chatExtrasStore.setChatInput({ token: this.token, text: '' })
+				this.chatExtrasStore.removeParentIdToReply(this.token)
```
Comment (DorraJaouad) [inline src/components/NewMessage/NewMessage.vue:520]: Add a check in `removeParentIdToReply()` for the existence of the token item in the store.

**Review by DorraJaouad — APPROVED** [review]
Tested 
- Notification 👍🏻 
- Reply with caption 👍🏻 
- Code review 👍🏻

File: src/components/NewMessage/NewMessage.vue (hunk 2 above)
Comment (Antreesy) [reply src/components/NewMessage/NewMessage.vue:520]: Vue has an inner guard for it, so nothing will happen if you call `Vue.delete(this.parentToReply, token)` and `this.parentToReply[token]` is undefined

**Comment by SystemKeeper** [comment]
This breaks rendering on talk-ios as a parent message within a file message was not possible until now. Please remember to test/align on these kind of changes :).
CC: @mahibi @rapterjet2004 To check on android as well

**Comment by nickvergessen** [comment]
> Please remember to test/align on these kind of changes

Sorry about that. That's right and I think we brought it up briefly in the chat/call. But should have raised a ticket right away.

**Pull request merged**